lib/OpenQA/Client/Archive.pm

```diff
@@ -32,7 +33,7 @@ sub run ($self, $options) {
     delete($job->{assets}->{repo});
 
     $path->child('testresults', 'thumbnails')->make_path if $options->{'with-thumbnails'};
-    $self->_download_assets($url, $job, $path);
+    $self->_download_assets($url, $job, $path) if $options->{'no-limit'};
```

Comment:
This is very misleading. `--no-limit` implies this is about the size. But the code no longer downloads assets by default.

Reply:
I suggest to clarify what behavior we want. The PR description is not clear right now.

For instance "Only download test results by default unless --download-assets is specified".

And maybe `--no-limit` should really only set `asset-size-limit=0`? That's what I would expect as a user without digging through the code.

Reply:
I also find this misleading. It would make sense to have an additional flag like `$options->{'no-download'}`. If you really want/need to change the default behavior than you could turn the flag around. Or you make it behave like this in case of `--asset-size-limit 0` as suggested by @kalikiana (which could still be combined with my suggestion for `--asset-size-limit infinity`).